pyresample/boundary/spherical_boundary.py

```diff
@@ -0,0 +1,177 @@
+    else:
+        return None, None
+
+
+def _is_clockwise_order(first_point, second_point, point_inside):
+    """Determine if polygon coordinates follow a clockwise path.
+
+    This uses :class:`pyresample.spherical.Arc` to determine the angle
+    between a polygon arc segment and a point known to be inside the polygon.
+
+    Note: pyresample.spherical assumes angles are positive if counterclockwise.
+    Note: if the longitude distance between the first_point/second_point and point_inside is
+    larger than 180°, the function likely return a wrong unexpected result !
+    """
+    from pyresample.spherical import Arc, SCoordinate
+    point1 = SCoordinate(*np.deg2rad(first_point))
+    point2 = SCoordinate(*np.deg2rad(second_point))
+    point3 = SCoordinate(*np.deg2rad(point_inside))
+    arc12 = Arc(point1, point2)
+    arc23 = Arc(point2, point3)
+    angle = arc12.angle(arc23)
+    is_clockwise = -np.pi < angle < 0
+    return is_clockwise
+
+
+def _check_is_clockwise(area, sides_x, sides_y):
+    from pyresample import SwathDefinition
+
+    if isinstance(area, SwathDefinition):
+        point_inside, start_idx = _try_get_local_inside_point(sides_x, sides_y)
+        first_point = sides_x[start_idx][0], sides_y[start_idx][0]
+        second_point = sides_x[start_idx][1], sides_y[start_idx][1]
+        return _is_clockwise_order(first_point, second_point, point_inside)
+    else:
+        if area.is_geostationary:
+            point_inside = area.get_lonlat(row=int(area.shape[0] / 2), col=int(area.shape[1] / 2))
+            first_point = sides_x[0][0], sides_y[0][0]
+            second_point = sides_x[0][1], sides_y[0][1]
+            return _is_clockwise_order(first_point, second_point, point_inside)
+        else:
+            return True
+
+
+class SphericalBoundary(BaseBoundary, OldBoundary):
+    """SphericalBoundary object.
+
+    The inputs must be the list of longitude and latitude boundary sides.
+    """
+    # NOTES:
+    # - Boundary provide the ancient method contour_poly and draw
+    #   from the old interface for compatibility to AreaBoundary
+
+    @classmethod
+    def _check_is_boundary_clockwise(cls, sides_x, sides_y, area):
+        """SphericalBoundary specific implementation."""
+        return _check_is_clockwise(area, sides_x, sides_y)
+
+    @classmethod
+    def _compute_boundary_sides(cls, area, vertices_per_side):
+        sides_lons, sides_lats = area._get_geographic_sides(vertices_per_side=vertices_per_side)
+        return sides_lons, sides_lats
+
+    def __init__(self, area, vertices_per_side=None):
+        super().__init__(area=area, vertices_per_side=vertices_per_side)
+
+        self.sides_lons = self._sides_x
+        self.sides_lats = self._sides_y
+
+        # Define CRS
+        if self.is_swath:
+            crs = self._area.crs
+        else:
+            crs = CRS(proj="longlat", ellps="WGS84")  # FIXME: AreaDefinition.get_lonlat for geographic projections?
```

Comment:
I was more thinking to another point to check ...  
The lon/lat coordinates that composes the `SphericalBoundary` sides are the coordinates that are then used for the spherical polygon intersection. 

It seems to me that we don't currently ensure that these lat/lon coordinates are based on a unique ellipsoid. 
The default ellipsoid for SwathDefinition is WGS84, but if one specify the crs with GRS80 at SwathDefinition, the lat/lon coordinates of `SwathDefinition.get_lonlats ` will be on GRS80, and in turns those of SphericalBoundary.
Similarly, `AreaDefinition.get_lonlats` returns coordinates on the ellipsoid of `area.crs.geodetic_crs`, which is not necessary WGS84. 
So I guess we are currently doing spherical operations with coordinates that are not defined on the same ellipsoid.

It's a "minor" problem, but maybe worth to consider or describe into a Issue.

Still I can still set `self._area.crs.geodetic_crs` here I guess ;)

Reply:
Yeah it might be a bit complicate to do the transform on the polygon class every time an operation must be performed. 
We can leave this for future PR, but I see three other options (in order of feasability): 
- we do the transform to WGS84 when creating the spherical polygon in `boundary().polygon`
- in the `SphericalBoundary` class, at creation, we force the lat/lon to be on unique ellipsoid i.e. WGS84 (or something settable with some configs) 
- we change the behaviour of  `area/swath get_lonlats` to return by default the WGS84.

Reply:
I'm not sure what "do the transform on the polygon class every time an operation must be performed" includes, but my hope is it wouldn't be too bad. And it can always be `projected boundary -> spherical boundary -> spherical polygon` or `projected boundary -> projected polygon -> spherical polygon` (maybe that's too ugly given the difference in polygon implementation) or as you mentioned in your first option I think it would be `projected boundary -> spherical polygon`.

I disagree with the last option of changing the default of get_lonlats. These are often used as the lon/lat version of the geometry definition and to change what coordinate system/datum they are on seems like the wrong move. Similarly, I'm scared of the spherical boundary transforming to a single ellipsoid in case people wanted to compare two boundaries of the same CRS. I could see maybe a `.normalize()` method on the boundary that ensures this single CRS, but...at that point I guess it would do the same as `.transform()` with a default CRS.